.github/workflows/nix.yml

```diff
@@ -0,0 +1,20 @@
+name: "Nix Test"
+
+# Controls when the workflow will run
+on:
+  # Trigger the workflow on push or pull request events:
+  push:
+    branches: [ master ]
+  pull_request:
+
+  # Allow to run this workflow manually from the Actions tab:
+  workflow_dispatch:
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: cachix/install-nix-action@v20
+    - run: nix build
+    - run: nix flake check
```

Comment:
Wenn wir hier auch noch die batou Tests ausführen können bin ich happy

Reply:
wenns in die ci rein soll macht der main workflow das auch schon

für das python paket wären die tests nett aber dauern auch recht lange

Reply:
update: in der nix sandbox schlagen einige tests fehl:

```=========================== short test summary info ============================
FAILED src/batou/lib/tests/test_archive.py::test_zip_extracts_archive_to_target_directory - batou.utils.CmdExecutionError: The deployment encountered an error.
FAILED src/batou/lib/tests/test_archive.py::test_zip_overwrites_existing_files - batou.utils.CmdExecutionError: The deployment encountered an error.
FAILED src/batou/lib/tests/test_buildout.py::test_runs_buildout_successfully - batou.utils.CmdExecutionError: The deployment encountered an error.
FAILED src/batou/lib/tests/test_file.py::test_directory_copies_all_files - batou.utils.CmdExecutionError: The deployment encountered an error.
FAILED src/batou/lib/tests/test_file.py::test_directory_file_copies_all_files - batou.utils.CmdExecutionError: The deployment encountered an error.
FAILED src/batou/lib/tests/test_file.py::test_directory_last_updated_reflects_file_changes - batou.utils.CmdExecutionError: The deployment encountered an error.
FAILED src/batou/lib/tests/test_file.py::test_directory_does_not_copy_excluded_files - batou.utils.CmdExecutionError: The deployment encountered an error.
FAILED src/batou/lib/tests/test_python.py::test_venv_updates_if_python_changes - batou.utils.CmdExecutionError: The deployment encountered an error.
FAILED src/batou/lib/tests/test_python.py::test_venv_does_not_update_if_python_does_not_change - batou.utils.CmdExecutionError: The deployment encountered an error.
FAILED src/batou/lib/tests/test_svn.py::test_runs_svn_to_clone_repository - batou.utils.CmdExecutionError: The deployment encountered an error.
FAILED src/batou/secrets/tests/test_editor.py::test_edit_file_has_secret_prefix - batou.GPGCallError: Exitcode 2 while calling: gpg --decrypt /private/tmp/ni...
FAILED src/batou/secrets/tests/test_editor.py::test_blank_edit - batou.GPGCallError: Exitcode 2 while calling: gpg --decrypt /private/tmp/ni...
FAILED src/batou/secrets/tests/test_manage.py::test_manage__1[add_user] - batou.GPGCallError: Exitcode 2 while calling: gpg -q --no-tty --batch --dec...
FAILED src/batou/secrets/tests/test_manage.py::test_manage__1[remove_user] - batou.GPGCallError: Exitcode 2 while calling: gpg -q --no-tty --batch --dec...
FAILED src/batou/secrets/tests/test_manage.py::test_manage__2 - AssertionError: assert '306151601E813A47' in ''
FAILED src/batou/secrets/tests/test_manage.py::test_manage__summary__1 - assert 1 is None
FAILED src/batou/secrets/tests/test_secrets.py::test_decrypt - batou.GPGCallError: Exitcode 2 while calling: gpg --decrypt /private/tmp/ni...
FAILED src/batou/secrets/tests/test_secrets.py::test_caches_cleartext - batou.GPGCallError: Exitcode 2 while calling: gpg --decrypt /private/tmp/ni...
FAILED src/batou/secrets/tests/test_secrets.py::test_write_should_fail_unless_write_locked - batou.GPGCallError: Exitcode 2 while calling: gpg --decrypt /private/tmp/ni...
FAILED src/batou/secrets/tests/test_secrets.py::test_write_unparsable_raises_error - batou.GPGCallError: Exitcode 2 while calling: gpg --decrypt /private/tmp/ni...
FAILED src/batou/secrets/tests/test_secrets.py::test_write - batou.GPGCallError: Exitcode 2 while calling: gpg --decrypt /private/tmp/ni...
FAILED src/batou/secrets/tests/test_secrets.py::test_write_fails_without_recipients - batou.GPGCallError: Exitcode 2 while calling: gpg --decrypt /private/tmp/ni...
FAILED src/batou/secrets/tests/test_secrets.py::test_write_fails_if_recipient_key_is_missing_keeps_old_file - batou.GPGCallError: Exitcode 2 while calling: gpg --decrypt /private/tmp/ni...
FAILED src/batou/secrets/tests/test_secrets.py::test_secrets_override_without_interpolation - batou.GPGCallError: Exitcode 2 while calling: gpg --decrypt environments/en...
FAILED src/batou/tests/test_deploy.py::test_main_with_errors - assert   batou/2.4a1.dev0 (cpython 3.10.11-final0, Darwin 22.5.0 arm64)
FAILED src/batou/tests/test_endtoend.py::test_example_errors_early - assert   batou/2.4a1.dev0 (cpython 3.10.11-final0, Darwin 22.5.0 arm64)
FAILED src/batou/tests/test_endtoend.py::test_example_errors_late - assert   batou/2.4a1.dev0 (cpython 3.10.11-final0, Darwin 22.5.0 arm64)
FAILED src/batou/tests/test_endtoend.py::test_diff_is_not_shown_for_keys_in_secrets - FileNotFoundError: [Errno 2] No such file or directory: 'work'
============ 28 failed, 391 passed, 1 skipped in 203.24s (0:03:23) =============
Exception ignored in: <_io.FileIO name=9 mode='rb+' closefd=True>
ResourceWarning: unclosed file <_io.BufferedRandom name=9>
py: exit 1 (204.22 seconds) /private/tmp/nix-build-python3.10-batou-latest.drv-0/source> pytest -vv pid=86473
.pkg: _exit> python /nix/store/m8lgpx7h0nmirhlwmld8kq8pi6yr5jr1-python3.10-pyproject-api-1.5.0/lib/python3.10/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
  py: FAIL code 1 (211.54=setup[7.32]+cmd[204.22] seconds)
  evaluation failed :( (211.74 seconds)

lines 3239-3311/3311 (END)
```